nemo_megatron_launcher/launch_scripts/nemo_megatron/collections/eval_harness/lm_eval/models/nemo_gpt3.py

```diff
@@ -0,0 +1,313 @@
+        # sanity check
+        assert len(context_enc) > 0
+        assert len(continuation_enc) > 0
+        assert len(continuation_enc) <= self.max_length
+
+        conti_len = len(continuation_enc)
+        inp_enc = torch.tensor((context_enc + continuation_enc)[-(self.max_length + 1) :])
+        return inp_enc, conti_len
+
+
+def setup_trainer_and_model(args):
+    """Setup model and optimizer."""
+    torch.set_grad_enabled(False)
+
+    assert args.nemo_model is not None or (
+        args.checkpoint_folder is not None and args.checkpoint_name is not None
+    ), "Path to checkpoints is required."
+
+    # cast precision to int if 32 or 16
+    if args.precision in ["32", "16"]:
+        args.precision = int(args.precision)
+
+    model_parallel_size = args.tensor_model_parallel_size * args.pipeline_model_parallel_size
+    num_nodes = max(model_parallel_size // 8, 1)
+    num_gpus = min(model_parallel_size, 8)
+    trainer = Trainer(
+        strategy=NLPDDPStrategy(), devices=num_gpus, num_nodes=num_nodes, precision=args.precision, accelerator='gpu'
+    )
+
+    app_state = AppState()
+    if args.tensor_model_parallel_size > 1 or args.pipeline_model_parallel_size > 1:
+        app_state.model_parallel_size = args.tensor_model_parallel_size * args.pipeline_model_parallel_size
+        (
+            app_state.tensor_model_parallel_rank,
+            app_state.pipeline_model_parallel_rank,
+            app_state.model_parallel_size,
+            *_,
+        ) = fake_initialize_model_parallel(
+            world_size=app_state.model_parallel_size,
+            rank=trainer.global_rank,
+            tensor_model_parallel_size_=args.tensor_model_parallel_size,
+            pipeline_model_parallel_size_=args.pipeline_model_parallel_size,
+        )
+
+    if args.nemo_model is not None:
+        logger.info(f"**** Loading checkpoint from {args.nemo_model}")
+        connector = SaveRestoreConnector()
+        if os.path.isdir(args.nemo_model):
+            connector._model_extracted_dir = args.nemo_model
+        model = MegatronGPTModel.restore_from(
+            restore_path=args.nemo_model, save_restore_connector=connector, trainer=trainer
+        )
+    else:
+        if args.tensor_model_parallel_size > 1 or args.pipeline_model_parallel_size > 1:
+            app_state.pipeline_model_parallel_size = args.pipeline_model_parallel_size
+            app_state.tensor_model_parallel_size = args.tensor_model_parallel_size
+
+        logger.info(f"**** Loading checkpoint from {args.checkpoint_folder} - {args.checkpoint_name}")
+        # inject model parallel rank
+        checkpoint_path = inject_model_parallel_rank(os.path.join(args.checkpoint_folder, args.checkpoint_name))
+        model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+
+    model.freeze()
+
+    return trainer, model
+
+
+def hacky_DDP_initialize(model):
```

Comment:
Let's not use "hacky" in our function name?

Reply:
dummy sounds better to you?